### PR TITLE
feat(itinerary-body): add and use showAlertEffectiveDateTimeText to alert components

### PR DESCRIPTION
**Pull request by josh-willis-arcadis**
Make the 'Effective as of [date]' line from the alert message configurable, as requested from the client.

### CODE REVIEW

File: packages/itinerary-body/src/TransitLegBody/alerts-body.tsx [hunk 1]
```diff
@@ -106,34 +108,35 @@ export default function AlertsBody({
...
-                      }
-                      description="Text with the date an alert takes effect"
-                      id="otpUi.TransitLegBody.AlertsBody.effectiveDate"
-                      values={{
-                        dateTime: effectiveStartDate * 1000
-                      }}
-                    />
-                  )}
+                  {showAlertEffectiveDateTimeText &&
+                    (Math.abs(dayDiff) <= 1 ? (
```
Comment (daniel-heppner-ibigroup) [inline packages/itinerary-body/src/TransitLegBody/alerts-body.tsx:112]: What happens if you wrap the parent component `S.TransitAlertEffectiveDate` with the new boolean? I think this would make it a bit clearer just because it would break up this bit conditional block here that can be a bit hard to follow. At first I thought it was `showAlertEffectiveDateTimeText && Math.abs(dayDiff) <= 1` because I didn't see the `(` so I think breaking it up would be good just for clarity.

File: packages/itinerary-body/src/types.ts [hunk 2]
```diff
@@ -201,6 +201,8 @@ interface ItineraryBodySharedProps {
   setViewedTrip: SetViewedTripFunction;
   /** If true, will show agency information in transit legs */
   showAgencyInfo?: boolean;
+  /** If true, will show the effect date/time text in the alert body */
```
Comment (daniel-heppner-ibigroup) [inline packages/itinerary-body/src/types.ts:204]: Since it's default to true, maybe change the comment to "if false..."

**Review by daniel-heppner-ibigroup — APPROVED** [review]
LGTM!

File: packages/itinerary-body/src/types.ts (hunk 2 above)
Comment (josh-willis-arcadis) [reply packages/itinerary-body/src/types.ts:204]: I agree. Good catch!

File: packages/itinerary-body/src/TransitLegBody/alerts-body.tsx (hunk 1 above)
Comment (josh-willis-arcadis) [reply packages/itinerary-body/src/TransitLegBody/alerts-body.tsx]: addressed in dc9fb59

**Review by amy-corson-ibigroup — APPROVED** [review]
LGTM! Thanks for including a story!!
